pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2824,3 +2824,44 @@ type ConfigFileVolumeSource struct {
 	// projected items for all in one resources secrets, configmaps, and downward API
 	Projected *v1.ProjectedVolumeSource `json:"projected,omitempty" protobuf:"bytes,26,opt,name=projected"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CephCOSIDriver represents the CRD for the Ceph COSI Driver Deployment
+// +kubebuilder:resource:shortName=cephcosi
+type CephCOSIDriver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	// Spec represents the specification of a Ceph COSI Driver
+	Spec CephCOSIDriverSpec `json:"spec"`
+}
+
+// CephCOSIDriverList represents a list of Ceph COSI Driver
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type CephCOSIDriverList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CephCOSIDriver `json:"items"`
+}
+
+// CephCOSIDriverSpec represents the specification of a Ceph COSI Driver
+type CephCOSIDriverSpec struct {
+	// Image is the container image to run the COSI driver
+	// +optional
+	Image string `json:"image,omitempty"`
+	// ObjectProviionerSideCarImage is the container image to run the COSI driver sidecar
+	// +optional
+	ObjectProvisionerSideCarImage string `json:"objectProvisionerSideCarImage,omitempty"`
+	// DeploymentStrategy is the strategy to use to deploy the COSI driver
+	// +optional
+	// +kubebuilder:validation:Enum=disabled;auto;force
```

Comment:
K8s API design uses capitalized values for string options.

```suggestion
	// +kubebuilder:validation:Enum=Disabled;Auto;Force
```